src/dtl/dyad_dtl_impl.c

```diff
@@ -0,0 +1,82 @@
+#include "dyad_dtl_impl.h"
+
+#include "ucx_dtl.h"
+#include "flux_dtl.h"
+
+dyad_rc_t dyad_dtl_init(dyad_dtl_t **dtl_handle,
+        dyad_dtl_mode_t mode, flux_t *h, bool debug)
+{
+    dyad_rc_t rc = DYAD_RC_OK;
+    *dtl_handle = malloc (sizeof (struct dyad_dtl));
+    if (*dtl_handle == NULL) {
+        return DYAD_RC_SYSFAIL;
+    }
+    (*dtl_handle)->mode = mode;
+    if (mode == DYAD_DTL_UCX) {
+        rc = dyad_dtl_ucx_init (
+            &((*dtl_handle)->private.ucx_dtl_handle),
+            mode,
+            h,
+            debug
+        );
+        if (DYAD_IS_ERROR(rc)) {
+            return rc;
+        }
+        (*dtl_handle)->rpc_pack = dyad_dtl_ucx_rpc_pack;
+        (*dtl_handle)->rpc_unpack = dyad_dtl_ucx_rpc_unpack;
+        (*dtl_handle)->rpc_respond = dyad_dtl_ucx_rpc_respond;
+        (*dtl_handle)->rpc_recv_response = dyad_dtl_ucx_rpc_recv_response;
+        (*dtl_handle)->establish_connection = dyad_dtl_ucx_establish_connection;
+        (*dtl_handle)->send = dyad_dtl_ucx_send;
+        (*dtl_handle)->recv = dyad_dtl_ucx_recv;
+        (*dtl_handle)->close_connection = dyad_dtl_ucx_close_connection;
```

Comment:
These should move into dtad_dtl_ucx_init()

Reply:
Actually, I'd argue they should stay here. The purpose of `dyad_dtl_ucx_init` (and its Flux RPC equivalent) is to do any setup and initialization of the underlying handle (e.g., `dyad_dtl_ucx_t`). On the other hand, the purpose of the top level initializer (`dyad_dtl_init`) is to initialize the "top-level" handle (i.e., `dyad_dtl_t`).

Reply:
Discussed over Slack and added